src/main/java/de/zalando/aruha/nakadi/partitioning/UserDefinedPartitioningStrategy.java

```diff
@@ -0,0 +1,31 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.PartitioningException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.List;
+
+public class UserDefinedPartitioningStrategy implements PartitioningStrategy {
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+            throws PartitioningException {
+        try {
+            final String partition = event.getJSONObject("metadata").getString("partition");
```

Comment:
I'd null check `event.getJSONObject("metadata")`. I know, you know that the partitioning strategy is applied after an event is validated against it's schema and that the schema currently requires the metadata field to be present. However, the interface of the function doesn't guarantee all that and I wouldn't rely on this.